accepted/00000-maintenance-windows.md

```diff
@@ -0,0 +1,208 @@
+- Feature Name: Maintenance Windows
+- Start Date: 2020-03-31
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Define Maintenance Windows in Uyuni and assign them to systems
+
+# Motivation
+[motivation]: #motivation
+
+Customers have company policies to perform changes on systems only at specific points in time.
+This is typically called "Maintenance Window".
+
+Uyuni should be able to define them and prevent executing of actions outside of such a window.
+
+# Architecture
+[architecture]: #architecture
+
+A system is in "Maintenance Mode" or not.
+When a system is in "Maintenance Mode" no action can be executed.
+Setting the Maintenance Mode "on" and "off" requires Organization Admin permissions.
+A "Maintenance Window" does nothing else than turning Maintenance Mode "on" when the window starts
+and "off" when it ends.
+
+Defining Maintenance Windows requires Organization Admin permissions.
+
+Maintenance Windows are defined on system level.
+Every System can get 1 Schedule assigned.
+A Schedule can be used for multiple systems.
+Inside of the Schedule, multiple Maintenance Windows can be defined.
+They can be defined as individual or as recurring events.
+
+To assign a Schedule to multiple systems, SSM should be used.
+
+
+# Detailed design
+[design]: #detailed-design
+
+## First Iteration
+
+In the first iteration we limit the implementation to load and update maintenance schedules
+and scheduling actions inside of maintenance windows.
+
+Enforcing Maintenance Windows on lower level is part of the future iterations and will be defined later
+in updates of this RFC.
```

Comment:
IIUC, this means that the implementation logic that actually blocks an action from being executed will be defined in future updates? I spoke with @hustodemon about Salt and traditional locking, and their conceptual differences. It might be a good occasion to unify the concept of "locking" in case we want to use it as a way to prevent action execution outside maintenance windows.